libs/wingsdk/src/cloud/bucket.ts

```diff
@@ -457,6 +469,21 @@ export interface IBucketClient {
    * @inflight
    */
   rename(srcKey: string, dstKey: string): Promise<void>;
+
+  /**
+   * Initiate a multipart upload to a given key in the bucket.
+   * @param key The key to upload to
+   * @returns The upload id for the multipart upload
+   * @inflight
+   */
+  multipartUpload(key: string): Promise<string>;
```

Comment:
Is the term "multipart" really helping here? I think the common developer won't really care that this is the underlying mechanism.

The docstrings should indicate that this will begin a multipart upload, but not needed in the name I think.

The return type can also be used to encode more meaning after you've landed on the right method. 

I would also return a struct (with the upload id inside) instead of a string so it's easier to correlate the start/stop.

So maybe something like this:

```js
struct MultipartUpload {
  id: str;
}

startUpload(key: str): MultipartUpload;

completeUpload(upload: MultipartUpload): void
```